src/cookie/Cookie.ts

```diff
@@ -0,0 +1,95 @@
+export class Cookie {
+    /**
+     * Get a Cookie
+     * @param {string} cookieName - The name / key of the Cookie to get.
+     * @returns {string|null} Returns the value of the Cookie, if this one was found. Otherwise return null.
+     */
+    public static getCookie(cookieName: string): string | null {
+        const match = document.cookie.match(new RegExp(`(^| )${encodeURIComponent(cookieName)}=([^;]+)`));
+        return match && decodeURIComponent(match[2]);
+    }
+
+    /**
+     * Set a Cookie
+     * @param {string} cookieName - Name of the Cookie to set.
+     * @param {string} cookieValue - Value of the Cookie to set.
+     * @param {Object} [options] - The differents options you can set on a cookie.
+     * @param {string} [options.path] - The path of the Cookie.
+     * If not specified, defaults to the current path of the current document location
+     * @param {string} [options.domain] - The domain the Cookie will be available on.
+     * If not specified, this defaults to the host portion of the current document location.
+     * @param {number} [options.maxAge] - The max age of the Cookie in seconds
+     * @param {number} [options.expires] - The expiration date of the Cookie, in GMTString format.
+     * See Date.toUTCString() for help formatting this value.
+     * @param {boolean} [options.secure] - This specify if the Cookie is only to be transmitted over secure protocols.
+     * @param {'lax'|'strict'} [options.samesite] - This prevents the browser from sending this Cookie along with
+     * cross-site requests.
+     * @returns {string} Returns the just created Cookie with his options
+     */
+    public static setCookie(
+        cookieName: string,
+        cookieValue: string,
+        options?: {
+            path?: string,
+            domain?: string,
+            maxAge?: number,
+            secure?: boolean,
+            expires?: string,
+            samesite?: 'lax'|'strict',
+        },
+    ) {
+        if (typeof cookieName !== 'string') throw new Error('cookieName must be a string');
+        if (typeof cookieValue !== 'string') throw new Error('cookieValue must be a string');
+
+        const cookie = [`${encodeURIComponent(cookieName)}=${encodeURIComponent(cookieValue)}`];
+
+        if (options) {
+            if (typeof options !== 'object') {
+                throw new Error('options must be an object');
+            }
+
+            if (options.path) {
+                if (typeof options.path === 'string') cookie.push(`path=${options.path}`);
+                else throw new Error('options.path must be a string');
```

Comment:
Usually, we do validation the other way around. Test for `!== string` and then throw. Only then write the value.
This also lets you simply write different validations underneath each other, removing the need to write them all in a single if() statement:
```suggestion
                if (typeof options.path !== 'string') throw new Error('options.path must be a string');
                cookie.push(`path=${options.path}`);
```

Reply:
I do agree this is "cleaner". But i still need to check if `options.path` exist, otherwise it will always throw an error if `options.path` is not defined.

Reply:
I just pushed something. Is it what you meant with your comment ?